gocourse03/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+const (
+	Chip       RodentType = "Chip"
+	Dale       RodentType = "Dale"
+	Cheaser    RodentType = "Cheaser"
+	Hackwrench RodentType = "Hackwrench"
+
+	NorthSector  Sector = "North"
+	SouthSector  Sector = "South"
+	WestSector   Sector = "West"
+	EastSector   Sector = "East"
+	CenterSector Sector = "Center"
+)
+
+type RodentType string
+
+type Sector string
+
+func main() {
+	var movements []Movement
+
+	rodents := []Rodent{
+		{ID: 1, RodentType: Chip, FromTo: [2]Sector{WestSector, WestSector}},
+		{ID: 2, RodentType: Dale, FromTo: [2]Sector{EastSector, EastSector}},
+		{ID: 3, RodentType: Cheaser, FromTo: [2]Sector{SouthSector, SouthSector}},
+		{ID: 4, RodentType: Hackwrench, FromTo: [2]Sector{NorthSector, NorthSector}},
+	}
+
+	westRodents := []Rodent{rodents[0]}
+	eastRodents := []Rodent{rodents[1]}
+	southRodents := []Rodent{rodents[2]}
+	northRodents := []Rodent{rodents[3]}
+
+	param := RodentParametr{
+		Rodents:      &rodents,
+		WestRodents:  &westRodents,
+		EastRodents:  &eastRodents,
+		SouthRodents: &southRodents,
+		NorthRodents: &northRodents,
+		Movements:    &movements,
+	}
+
+	PrintLocations(&rodents, &movements)
+
+	moveThroughCenter(1, time.Now().Add(10), WestSector, EastSector, param)
+	moveThroughCenter(2, time.Now().Add(11), EastSector, NorthSector, param)
+	moveThroughCenter(2, time.Now().Add(12), NorthSector, SouthSector, param)
+	moveThroughCenter(3, time.Now().Add(13), SouthSector, WestSector, param)
+
+	PrintLocations(&rodents, &movements)
```

Comment:
Спробуй змінити функції `PrintLocations` та `moveThroughCenter`, щоб працювало так:
```suggestion
	var movements []Movement
	PrintLocations(rodents, movements)

	movements = moveThroughCenter(1, time.Now().Add(10), WestSector, EastSector, param, movements)
	movements = moveThroughCenter(2, time.Now().Add(11), EastSector, NorthSector, param, movements)
	movements = moveThroughCenter(2, time.Now().Add(12), NorthSector, SouthSector, param, movements)
	movements = moveThroughCenter(3, time.Now().Add(13), SouthSector, WestSector, param, movements)

	PrintLocations(rodents, movements)
```

Тоді можна видалити буде поле `Movements *[]Movement` з `RodentParameter` і код стане більш ідіоматичним.